automation/tdx_canonical_verifier.sh

```diff
@@ -0,0 +1,228 @@
+#!/bin/bash
+CUR_DIR=$(pwd)
+TDX_CONFIG=tdx-config
+TDX_DIR="$CUR_DIR"/tdx_verifier
+source "$CUR_DIR"/tdx_canonical_common
+source "$CUR_DIR"/$TDX_CONFIG
+GUEST_TOOLS_DIR="$TDX_DIR"/guest-tools
+GUEST_IMG_DIR="$GUEST_TOOLS_DIR"/image
+TD_GUEST_VERIFY_TEXT="tdx: Guest detected"
+TDT_HOST_VERIFY_TEXT="tdx: module initialized"
+SERVICE_ACTIVE_STATUS_VERIFY_TEXT="Active: active (running)"
+MPA_SERVICE_CHECK="mpa_registration_tool.service; enabled; preset: enabled"
+VERIFY_ATTESTATION_SCRIPT="verify_attestation_td_guest.sh"
+TD_GUEST_PASSWORD=123456
+TD_GUEST_PORT=10022
+MPA_REGISTRATION_CHECK="INFO: Registration Flow - Registration status indicates registration is completed successfully"
+TRUSTAUTHORITY_API_FILE="config.json"
+PASSED=PASSED

```

Comment:
Logic to determine fail/pass needs to change. Let's discuss more on this. Not liking this approach of setting so many variables as FAILED and then setting them as PASSED.

Reply:
handle this implementation using finally block